lib/matplotlib/axes/_axes.py

```diff
@@ -2998,15 +3021,30 @@ def extract_err(err, data):
             if xlolims.any():
                 yo, _ = xywhere(y, right, xlolims & everymask)
                 lo, ro = xywhere(x, right, xlolims & everymask)
-                barcols.append(self.hlines(yo, lo, ro, **eb_lines_style))
+
+                if border is not None:
+                    barcols.append(self.hlines(yo, lo, ro,
+                                               **eb_lines_style)
+                                               .set_path_effects(path_effects))
+                else:
+                    barcols.append(self.hlines(yo, lo, ro, **eb_lines_style))
+
                 rightup, yup = xywhere(right, y, xlolims & everymask)
                 if self.xaxis_inverted():
                     marker = mlines.CARETLEFTBASE
                 else:
                     marker = mlines.CARETRIGHTBASE
-                caplines.append(
-                    mlines.Line2D(rightup, yup, ls='None', marker=marker,
-                                  **eb_cap_style))
+
+                if border is not None:
```

Comment:
I think all of these changes could be avoided by adding `path_effects` to the `eb_cap_style` and `eb_lines_style` dictionaries?